spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/AmqpMessageHeaderErrorMessageStrategy.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.amqp.support;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.springframework.core.AttributeAccessor;
+import org.springframework.integration.support.ErrorMessageStrategy;
+import org.springframework.integration.support.ErrorMessageUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.ErrorMessage;
+
+/**
+ * {@link ErrorMessageStrategy} extension that adds the raw AMQP message as
+ * a header to the {@link org.springframework.integration.message.EnhancedErrorMessage}.
+ *
+ * @author Gary Russell
+ * @since 4.3.10
+ *
+ */
+public class AmqpMessageHeaderErrorMessageStrategy implements ErrorMessageStrategy {
+
+	/**
+	 * Header name/retry context variable for the raw received message.
+	 */
+	public static final String AMQP_RAW_DATA = "amqp_data"; // move to AmqpHeaders.RAW_DATA in 2.0
```

Comment:
BTW, are you going to push commit to SA for this `MESSAGE` constant?
I mean, if we are going to leverage it there, this is a blocker for backporting this change to SI `master`.